ffcx/codegeneration/coordinate_mapping.py

```diff
@@ -204,7 +204,7 @@ def compute_physical_coordinates(L, ir):
 
 def compute_reference_geometry(L, ir):
     degree = ir.coordinate_element_degree
-    if degree == 1:
+    if degree == 1 and ir.cell_shape in ("interval", "triangle", "tetrahedron"):
```

Comment:
We should probably prepare a more standard way of telling if a coordinate map is affine, maybe have `ir.affine = True`?